Assets/Scripts/Combat/PlayerCombat.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerCombat : Entity {
+
+    [SerializeField, Tooltip("The range of the player attack as a float")]
+    float _attackRange;
+    [SerializeField, Tooltip("How often the player can attack")]
+    float _attackRate;
+
+    float _nextAttackTime = 0f;
+
+    public Animator animator;
+    public Transform attackPoint;
+    public LayerMask enemyLayers;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (Time.time >= _nextAttackTime) {
+            if (Input.GetKeyDown(KeyCode.X)) {
+                Attack();
+                _nextAttackTime = Time.time + 1f / _attackRate;
+            }
```

Comment:
Consider checking out using Time.deltaTime for counting down attack time. 
If you subtract Time.deltaTime from a varialbe it will decrease by exactly 1 per second.
So in this example:

```csharp
float _cooldown = 5f;

void Update () {
    _cooldown -= Time.deltaTime;
}
```
... cooldown will be 0 after 5 seconds.